### ISSUE TITLE
Could for have a warning for attribute shadowing in C++?

### ISSUE THREAD

**Issue by abique** [llvm/llvm-project#71405]
Hi,

The following C++ snippet doesn't trigger a single warning when compiled with `-Wall -Wextra -pedantic`:
```C++
struct A
{
  int x;
};

struct B : public A
{
  int x;
};
```

Do you think it'd be reasonable to offer a warning for detecting it?
Tested with clang 16.0.6.

Thank you very much.
